application/common/helpers/MySqlDateTime.php

```diff
@@ -35,6 +35,7 @@ public static function formatDateTime(int $timestamp): string
      */
     public static function relative(string $difference = '+30 days'): string
     {
+        $difference = str_replace('--', '+', $difference);
```

Comment:
This fixed an email test where it was being passed `-2 days` and strtotime started to be unable to handle `--2 days` in our php8.3, but it worked in our php8.1.

Reply:
I've tried adjusting it in the [test](https://github.com/silinternational/idp-id-broker/blob/main/application/features/bootstrap/UnitTestsContext.php#L468), but nothing worked. I thought about adjusting it [here](https://github.com/silinternational/idp-id-broker/blob/main/application/common/models/User.php#L1398), but what I did now seemed a better place to fix it.

Reply:
OK, I guess it can stay this way. Thanks.

Reply:
Though I wonder if it would be better to do a check to make sure it begins with the `--` instead of just a complete str_replace.

```
if (substr($difference, 0, 2) == '--') {
    $difference = substr($difference, 2);
} 
```

Reply:
Or use trim instead of replace

Reply:
`trim` I think only is characters, not strings.

Reply:
It all starts with the app parameter being the negative of what it should be, because there's a weird logical relationship between the name of the parameter and how it is used. Perhaps an sprintf of the * -1 would be better?